time.rkt

```diff
@@ -191,16 +331,61 @@
     (fprintf port "<section id='profile'><h1>Profiling</h1>")
     (fprintf port "<p class='load-text'>Loading profile data...</p></section>")))
 
-(define (run test-id p)
+(define (run test-id p timeline-port)
   (define operation-table
     (and (or (not test-id) (not (string->number test-id))) (make-operation-table test-id)))
   (define-values (expression-table expression-footer)
     (if (and p (or (not test-id) (string->number test-id)))
-        (make-expression-table p test-id)
+        (make-expression-table p test-id timeline-port)
         (values #f #f)))
   (list operation-table expression-table expression-footer))
 
-(define (generate-html html-port profile-port operation-table expression-table expression-footer)
+(define (generate-ratio-plot dir)
+  (define-values (sp out in err)
+    (subprocess #f #f #f (find-executable-path "python3")
+                "infra/ratio_plot.py"
+                "-t" (format "~a/timeline.json" dir)
+                "-o" (format "~a/ratio_plot.png" dir)))
+  (printf "~a" (port->string out)) ; macros for latex
+  (close-input-port out)
+  (close-output-port in)
+  (close-input-port err)
+  (subprocess-wait sp))
```

Comment:
I would rather this be called in `nightly.sh` instead of here

Reply:
(and same for the other plots)

Reply:
(This should provide better / built-in logging if it fails.)

Reply:
Okay, sure